### ISSUE TITLE
Can't define instances when running web interface

### ISSUE THREAD

**Issue by danieldjohnson** [google-research/dex-lang#202]
This works when run with `dex script` but errors in `dex web`:

```
data Foo a:Type = MkFoo Int

@instance fooInt : Foo Int = MkFoo 42

def theFoo (x:Foo Int)?=> : Int = case x of MkFoo x -> x
:p theFoo
> Type error:Couldn't synthesize a class dictionary for: (Foo Int64)
>
> :p theFoo
>    ^^^^^^
```

**Comment by dougalm** [comment]
That's bad but not surprising. We form a DAG of dependencies [here](https://github.com/google-research/dex-lang/blob/main/src/lib/LiveOutput.hs#L89) based on free variables and variable definitions. But we should unconditionally assume a dependence on all instance definitions.

We should probably also add some tests for the web interface.The whole thing is a bit janky. I'm hoping one day we can restructure things to be more like a [query oriented compiler](https://ollef.github.io/blog/posts/query-based-compilers.html), and the incremental notebook will just fall out from generic caching.

**Comment by apaszke** [comment]
I've fixed this by breaking the DAG creation in #262.
